elastic/kibana
# [Index Management] Fix data streams API integration test

## Summary

Fix https://github.com/elastic/kibana/issues/69093

### How to test

Current error with ES can be reproduced by running ES from source or somehow getting a build with the changes referenced in the above issue.

Send the following request to the cluster (password is just default `elastic:password`):

```sh
curl --request PUT \
  --url http://localhost:9200/_index_template/test-data-stream \
  --header 'authorization: Basic ZWxhc3RpYzpwYXNzd29yZA==' \
  --header 'content-type: application/json' \
  --data '{
	"index_patterns": [
		"*"
	],
	"template": {
		"settings": {}
	},
	"data_stream": {
		"timestamp_field": "@timestamp"
	}
}'
```

And the issue reported in the error should surface. Running the same request with the alterations to the body proposed here should fix the request for creating the composable index template.

### Additional context

Fix inspired by ES docs https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-create-data-stream.html
